### PR TITLE
VendorInfo unit tests

### PR DESCRIPTION
**Pull request by DenisRybas**

### CODE REVIEW

File: x/vendorinfo/types/messages_vendor_info_test.go [hunk 1]
```diff
@@ -18,13 +18,19 @@ func TestMsgCreateVendorInfo_ValidateBasic(t *testing.T) {
...
+				VendorName:       testconstants.VendorName,
+				CompanyLegalName: testconstants.CompanyLegalName,
 			},
 			err: sdkerrors.ErrInvalidAddress,
 		}, {
 			name: "valid address",
 			msg: MsgCreateVendorInfo{
-				Creator: sample.AccAddress(),
+				Creator:          sample.AccAddress(),
+				VendorID:         int32(testconstants.VendorID1),
```
Comment (ashcherbakov) [inline x/vendorinfo/types/messages_vendor_info_test.go:31]: Makes sense to add more tests here to check validation of messages as defined in https://github.com/zigbee-alliance/distributed-compliance-ledger/blob/issue-99-cosmos-upgrade-v0.44/proto/vendorinfo/tx.proto:
- min/max value
- check that VID is unit16 ([0, 65535])

File: x/vendorinfo/handler_test.go [hunk 2]
```diff
@@ -15,280 +15,278 @@
...
+		dclauthtypes.Trustee,
+		dclauthtypes.NodeAdmin,
+	} {
+		accAddress := GenerateAccAddress()
+		setup.AddAccount(accAddress, []dclauthtypes.AccountRole{role}, setup.VendorID)
+
+		// update existing vendorinfo by user without Vendor role
+		msgUpdateVendorInfo := NewMsgUpdateVendorInfo(accAddress)
+		_, err = setup.Handler(setup.Ctx, msgUpdateVendorInfo)
+		require.NoError(t, err)
```
Comment (ashcherbakov) [inline x/vendorinfo/handler_test.go:198]: So, a use without a Vendor role can update existing VendorInfo.... Is it expected? I think no.

File: utils/validator/errors.go [hunk 3]
```diff
@@ -28,4 +28,5 @@ var (
 	ErrFieldNotValid           = sdkerrors.Register(Codespace, 902, "field not valid")
 	ErrFieldLowerBoundViolated = sdkerrors.Register(Codespace, 903, "field lower bound violated")
 	ErrFieldUpperBoundViolated = sdkerrors.Register(Codespace, 904, "field upper bound violated")
+	ErrFieldMinLengthExceeded  = sdkerrors.Register(Codespace, 905, "field min length not reached")
```
Comment (spivachuk) [inline utils/validator/errors.go:31]: It would be better to rename it to `ErrFieldMinLengthNotReached`.

File: x/vendorinfo/genesis_test.go [hunk 4]
```diff
@@ -3,12 +3,39 @@ package vendorinfo_test
...
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
 	keepertest "github.com/zigbee-alliance/distributed-compliance-ledger/testutil/keeper"
+	dclauthtypes "github.com/zigbee-alliance/distributed-compliance-ledger/x/dclauth/types"
 	"github.com/zigbee-alliance/distributed-compliance-ledger/x/vendorinfo"
 	"github.com/zigbee-alliance/distributed-compliance-ledger/x/vendorinfo/types"
 )
 
+type DclauthKeeperMock struct {
```
Comment (spivachuk) [inline x/vendorinfo/genesis_test.go:15]: Can we define such mock type within `vendorinfo` cosmos module only once?

File: x/vendorinfo/types/messages_vendor_info_test.go (hunk 1 above)
Comment (DenisRybas) [reply x/vendorinfo/types/messages_vendor_info_test.go]: done

File: x/vendorinfo/handler_test.go (hunk 2 above)
Comment (DenisRybas) [reply x/vendorinfo/handler_test.go]: fixed

File: utils/validator/errors.go (hunk 3 above)
Comment (DenisRybas) [reply utils/validator/errors.go]: fixed

File: x/vendorinfo/genesis_test.go (hunk 4 above)
Comment (DenisRybas) [reply x/vendorinfo/genesis_test.go]: fixed

**Review by ashcherbakov — APPROVED** [review]

**Review by spivachuk — APPROVED** [review]

File: x/vendorinfo/types/messages_vendor_info_test.go [hunk 5]
```diff
@@ -1,104 +1,303 @@
...
+				VendorID:             testconstants.VendorID1,
+				VendorName:           testconstants.VendorName,
+				CompanyLegalName:     testconstants.CompanyPreferredName,
+				VendorLandingPageURL: "ABC",
+			},
+			err: validator.ErrFieldNotValid,
+		},
+		{
+			name: "vendor landing page URL len > 256",
+			msg: MsgCreateVendorInfo{
```
Comment (ashcherbakov) [inline x/vendorinfo/types/messages_vendor_info_test.go:123]: Makes sense to add a couple of more negative tests:
1) VendorName is is not set (empty string). Expected: ErrRequiredFieldMissing
2) CompanyLegalName is is not set (empty string). Expected: ErrRequiredFieldMissing

**Pull request merged**